x/feegrant/keeper/keeper.go

```diff
@@ -49,6 +50,25 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 
 	store := ctx.KVStore(k.storeKey)
 	key := feegrant.FeeAllowanceKey(granter, grantee)
+
+	var exp *time.Time
+	existingGrant, err := k.getGrant(ctx, grantee, granter)
+	if err != nil && existingGrant.GetAllowance() != nil {
+		grant1, err := existingGrant.GetGrant()
+		if err != nil {
+			return err
+		}
+
+		exp, err = grant1.ExpiresAt()
+		if err != nil {
+			return err
+		}
+
+		if exp != nil {
+			k.removeFromGrantQueue(ctx, feegrant.FeeAllowancePrefixQueue(exp, key))
+		}
+	}
```

Comment:
When do we really want to overwrite a grant? 

Maybe we should always overwrite a grant? Then we don't need to remove anything above.

Reply:
we should remove from the queue even if we overwrite the grant all the times. we should make new key from `expiration` and we need to remove the old key.

Reply:
so, imo, we should only remove+insert only when the expiration changed. We should not overwrite otherwise.